enhancements/custom-alert-configuration.md

```diff
@@ -0,0 +1,439 @@
+do not conform to the standard use case. We would then subtract this from the
+rest of the PDBs at limit to stop alerting on the KubeVirt ones.
+
+The metric should have at least a namespace and poddisruptionbudget name.
+
+The alert query could look like this:
+
+```json
+max by (namespace, poddisruptionbudget) (kube_poddisruptionbudget_status_current_healthy >= kube_poddisruptionbudget_status_desired_healthy and on (namespace, poddisruptionbudget) (kube_poddisruptionbudget_status_expected_pods > 0) unless on(namespace, poddisruptionbudget) (ocp_kubevirt_vmi_owned_pdbs))
+```
+
+Kubevirt would maintain/own `ocp_kubevirt_vmi_owned_pdbs` metric and ensure
+only the valid PDBs are ignored. This should prevent users from inserting their
+own PDBs into this ignore list.
+
+The disadvantage is that it is not scalable for more components. The metric
+describes only KubeVirt use case.
+
+### Silencing the alert
+
+It is possible to partially silence alerts that are firing either manually via
+an [openshift console][7] or via an [alertmanager API][8].
+
+It can filter out alerts according to known prometheus labels. Here we check
+for the poddisruptionbudget label with a `kubevirt-disruption-budget-.*`  value.
+
+```json
+curl -s -X POST -H "Authorization: Bearer ${TOKEN}" -H 'Accept: application/json' -H 'Content-Type: application/json' "${ALERTMANANER_URL}/api/v2/silences" -d '{
+	"status": {
+		"state": "active"
+},
+	"comment": "kubevirt PDBs",
+	"createdBy": "kube:admin",
+	"startsAt": "2023-01-12T20:37:30.995Z"
+	"endsAt": "2024-01-12T22:32:07.000Z",
+	"matchers": [
+  	{
+    	"isRegex": false,
+        "isEqual": true,
+    	"name": "alertname",
+    	"value": "PodDisruptionBudgetAtLimit"
+  	},
+  	{
+    	"isEqual": true,
+    	"isRegex": true,
+    	"name": "poddisruptionbudget",
+    	"value": "kubevirt-disruption-budget-.*"
+  	},
+  	{
+    	"isEqual": true,
+    	"isRegex": false,
+    	"name": "severity",
+    	"value": "warning"
+  	}
+	],
+  },
+'
+```
+
+This would work in a sense that it would stop notifications for this alert and
+would be silenced in the UI.
+
+There might be following problems with this approach:
+- In the UI it shows the alert is firing and being silenced at the same time.
+  It might prompt users to action when seeing the alert is silenced. When you
+  discover the alert via an API it just shows this alert is firing, and you
+  need to match this to silences to get the whole picture.
+- The silence is meant as a temporary solution and there needs to be `endsAt`
+  date and time defined. It is also possible to expire the silence with one
+  click in the UI. We would need to manage the silence to be sure it is always
+  present and not expired.
+- As mentioned above this is not a standard solution, and we are not shipping
+  any silences with openshift AFAIK. We would probably need a controller that
+  would reconcile the alert through alertmanager API.
```

Comment:
"reconcile the alert" -> "reconcile the silence"?

And user-workload monitoring has [a v1beta1 AlertmanagerConfig][1] now, which is starting to tiptoe into the "federated Alertmanaged config via Kube objects" space, so it's possible that Silences-as-Kube-objects aren't impossibly far away.

[1]: https://docs.openshift.com/container-platform/4.12/monitoring/managing-alerts.html#creating-alert-routing-for-user-defined-projects_managing-alerts

Reply:
good to know